cloud-services/mbl-cloud-client/scripts/arm_update_activate.sh

```diff
@@ -194,6 +275,45 @@ if echo "${FIRMWARE_FILES}" | grep .ipk$; then
 
     exit "$?"
 
+elif BOOTLOADER_FILES=$(echo "${FIRMWARE_FILES}" | grep "${WKS_BOOTLOADER_FILENAME_RE}"); then
+
+    # ------------------------------------------------------------------------------
+    # Write one or more bootloader components from an update payload file to disk
+    # ------------------------------------------------------------------------------
+
+    # variable is unquoted to remove carriage returns, tabs, multiple spaces etc
+    for bl_file in $BOOTLOADER_FILES; do
+        bl_filename_no_suffix=$(echo "${bl_file}" | sed 's/.gz//')
```

Comment:
I suggest:
* using `printf`(for reasons stated earlier);
* escaping the `.` in the regex;
* adding a `$` to the end of the regex so that it only matches at the end of the string.
```suggestion
        bl_filename_no_suffix=$(printf "%s\n" "$bl_file" | sed 's/\.gz$//')
```